### DISCUSSION TITLE
The library is imported into Next.js, and it runs fine locally, but after building, it throws an error

### DISCUSSION THREAD

**Discussion by huadee1108** [gilbarbara/react-joyride#933] [General]
## 🐛 Bug Report

![image](https://github.com/gilbarbara/react-joyride/assets/33452487/ffd5d9a8-7b57-4d09-9347-2317b884559f)
![image](https://github.com/gilbarbara/react-joyride/assets/33452487/b717f8e5-44a0-42a4-b631-f0582c80868c)
![image](https://github.com/gilbarbara/react-joyride/assets/33452487/94c0dcc7-6dfb-42fd-961e-5d2900501534)
![image](https://github.com/gilbarbara/react-joyride/assets/33452487/ea00db7f-53d3-4d02-a933-b7da921ffae2)

**Comment by gilbarbara** [comment]
Hey @huadee1108 

Based on the error image, this seems like an implementation error, not a library bug.
It's hard to suggest anything without additional details from your setup, like the Next.js version, and if you are using the `app` directory.
I'm moving this to discussions.

**Reply by huadee1108** [reply]
The 'app' directory of Next.js is not being used. It runs fine in the development environment, but after building, the error mentioned above occurs. I've reviewed the code, and it's likely originating from within the Joyride component.

**Reply by gilbarbara** [reply]
`Cannot read properties of undefined (reading '0')` doesn't mean much. 
This library only uses forEach in a single function and it doesn't try to access the index (0) directly...

Without a reproducible example, there's not much I could do.

**Reply by huadee1108** [reply]
![image](https://github.com/gilbarbara/react-joyride/assets/33452487/a199a44b-7bdf-4c0a-8c5b-7e691af0a9ee)
dev
![image](https://github.com/gilbarbara/react-joyride/assets/33452487/1356d107-038b-4d6d-ab14-2a0208d81eb8)
after build
![image](https://github.com/gilbarbara/react-joyride/assets/33452487/3a20b479-aced-4e49-9cc0-0b839125f376)
console no errors

I don't know how to pinpoint the issue now, as the error occurs after the build. I can't identify which line of code is causing the error.

**Reply by gilbarbara** [reply]
I've just added Joyride v2.5.5 in a next.js v13.4.x app and it was built and ran without errors.
Try removing some steps, and remove the custom Tooltip component. Simplify the setup.
If you still need help, create a minimal reproducible repo.

**Reply by huadee1108** [reply]
![image](https://github.com/gilbarbara/react-joyride/assets/33452487/9f33b65b-f1f6-4d0d-a341-05d77e94ce82)
The error only occurs when there's a 'steps' field present; it won't trigger if 'steps' is []. So, the point of error should be when iterating through the 'steps'.

I use next@13.0.2  react-joyride^2.5.5

**Reply by Sky-FE** [reply]
Hey @huadee1108 
I also encountered this problem now. How did you solve it in the end?

**Comment by billnbell2** [comment]
turn off swcMinify
